synapseclient/models/mixins/access_control.py

```diff
@@ -500,67 +517,104 @@ async def main():
-            )
-            return
+        custom_message = "Deleting ACLs [Dry Run]..." if dry_run else "Deleting ACLs..."
+        with shared_download_progress_bar(
+            file_size=1, synapse_client=client, custom_message=custom_message, unit=None
+        ) as progress_bar:
+            progress_bar.update(1)  # Initial setup complete
+
+            if should_process_children:
+                if recursive and not include_container_content:
+                    raise ValueError(
+                        "When recursive=True, include_container_content must also be True. "
+                        "Setting recursive=True with include_container_content=False has no effect."
+                    )
 
-        if include_self:
-            await self._delete_current_entity_acl(
-                client=client,
-                dry_run=dry_run,
-                benefactor_tracker=benefactor_tracker,
-            )
+                progress_bar.total += 1
+                progress_bar.refresh()
 
-        if should_process_children:
-            if include_container_content:
-                await self._process_container_contents(
+                all_entities = await self._collect_entities(
                     client=client,
                     target_entity_types=normalized_types,
-                    dry_run=dry_run,
+                    include_container_content=include_container_content,
+                    recursive=recursive,
+                    progress_bar=progress_bar,
+                )
+                progress_bar.update(1)
+
+                entity_ids = [entity.id for entity in all_entities if entity.id]
+                if entity_ids:
+                    progress_bar.total += 1
+                    progress_bar.refresh()
+                    await benefactor_tracker.track_entity_benefactor(
+                        entity_ids=entity_ids,
+                        synapse_client=client,
+                        progress_bar=progress_bar,
+                    )
+                else:
+                    progress_bar.total += 1
+                    progress_bar.refresh()
+                    progress_bar.update(1)
+
+            if is_top_level:
+                progress_bar.total += 1
+                progress_bar.refresh()
+                await self._build_and_log_run_tree(
+                    client=client,
                     benefactor_tracker=benefactor_tracker,
+                    collected_entities=all_entities,
+                    include_self=include_self,
+                    show_acl_details=show_acl_details,
+                    show_files_in_containers=show_files_in_containers,
+                    progress_bar=progress_bar,
+                    dry_run=dry_run,
                 )
 
-            if recursive and hasattr(self, "folders"):
-                await self._process_folder_permission_deletion(
+            if dry_run:
+                return
```

Comment:
dry_run returns here, so I was able to remove passing the boolean to the methods called after this